core/src/toga/widgets/numberinput.py

```diff
@@ -31,7 +31,11 @@ def _clean_decimal(value, step=None):
     value = Decimal(value)
 
     if step is not None:
-        value = value.quantize(step, rounding=ROUND_DOWN)
+        # ROUND_DOWN would not be unreasonable for manually-entered strings, but it
+        # interacts badly with native increment operations that use floats. For example,
+        # 1.2 might be incremented to 1.299999997, and then rounded back down to 1.2,
+        # so nothing would change.
```

Comment:
Good point - but I guess my question is why HALF_UP rather than HALF_DOWN? I guess it doesn't *really* matter, as it's literally only the 1.500000000 case that matters - but HALF_DOWN would seem more consistent with the ROUND_DOWN in that case.

Reply:
This only makes a difference when the app user manually enters a number that ends with a 5, so we should do whatever will confuse the smallest number of those users, who are likely not to have any advanced mathematics education.

ROUND_HALF_DOWN is not commonly used by anyone, so the real choice is between ROUND_HALF_UP (which I was taught in school), and ROUND_HALF_EVEN (which @freakboy3742 was taught in school).

ROUND_HALF_EVEN is used by the Python `round` function, and in most scientific contexts.

ROUND_HALF_UP is used by spreadsheets, and, based on my searches, in the majority of basic mathematics education, at least in English. For example, see "as taught in school" in the [AppleScript documentation](https://developer.apple.com/library/archive/documentation/AppleScript/Conceptual/AppleScriptLangGuide/reference/ASLR_cmds.html#//apple_ref/doc/uid/TP40000983-CH216-SW43).

So I think we should go with ROUND_HALF_UP.

Reply:
That reasoning all makes sense to me. (also - lol at AppleScript having an actual "as taught in school" keyword....)